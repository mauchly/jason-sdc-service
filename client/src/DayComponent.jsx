import React from 'react';
import { zeroPadding } from './helperFunc.js'

var DayComponent = (props) => {
  var day = props.days;
  var month = zeroPadding(props.month);
  var value = `${month}-${day}`;
  if (day !== '') {
    value = `${month}-${day}`;
  } else {
    value = 'empty'
  }
  var name;
  console.log('newBookedRange', props.newBookedDateRange)
  if (props.newBookedDateRange.includes(value)){
    console.log('wentIn newbookedDates')
    name = 'bookedDates';
  }
  if (!props.booked.includes(value)) {
   name = 'availableDay'
  } else {
    name = 'unavailableDay'
  }
  if (value === 'empty') {
    name = 'empty'
  }
  return (

<td className={name} onClick={props.onDayClick} id={value}>{props.days}</td>

  )

}

export default DayComponent;