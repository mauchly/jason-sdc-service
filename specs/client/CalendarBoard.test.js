import React from 'react';
import Enzyme from 'enzyme';
import 'babel-polyfill';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, mount, render } from 'enzyme';
import  DayComponent  from '../../client/src/DayComponent.jsx';

Enzyme.configure({ adapter: new Adapter() });

jest.mock('../../client/src/DayComponent.jsx', () => 'DayComponent');

const mockFn = jest.fn();

// const wrapper = mount(<DayComponent className="availableDay" onDayClick={mockFn}/>);

describe('DayComponenet Component', () => {

  it('should be defined', () => {
    expect(DayComponent).toBeDefined();
  });

  it('should render correctly', () => {
    const wrapper = shallow(
      <DayComponent />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should call mock function when a day is clicked', () => {

    const wrapper = shallow(<DayComponent className="availableDay" onDayClick={mockFn} id='3-31'/>);

   wrapper.find('.availableDay').simulate('click')

    expect(wrapper.find('.availableDay').length).toEqual(1);
    expect(wrapper.props().id).toEqual('3-31')
  })

})