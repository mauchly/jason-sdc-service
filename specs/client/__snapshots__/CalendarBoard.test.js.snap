// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalendarBoard Component should render correctly 1`] = `ShallowWrapper {}`;

exports[`DayComponenet Component should render correctly 1`] = `ShallowWrapper {}`;

exports[`elements contained in CalendarBoard component should render correctly 1`] = `ShallowWrapper {}`;
